### PR TITLE
Feature/wpandroid database downgrade

### PR DESCRIPTION
**Pull request by nbradbury**
In the past the app would crash after we switched to a branch that used a previous db version. Getting around that required clearing storage or uninstalling, and then logging into the app again. I found myself often doing that multiple times a day so I made fixing it my hack week project.

The solution required creating a WellSqlConfig descendant which overrides `onDowngrade()`, and uses that to drop & recreate all the tables. This is only enabled for debug users when the following has been added to `gradle.properties`:

```
wp.reset_db_on_downgrade = true
```

So now when a downgrade is detected, you'll see a red toast alerting that the db has been reset, and you'll be sent to the My Site screen. Until the sites are loaded you'll see a message about how you don't have any sites, but once sites are loaded the My Site page populates correctly. I won't claim it's a great experience, but since it's just for us I figured it was okay :)

To test, I recommend adding this to WPWellSqlConfig:

```
override fun getDbVersion(): Int {
    return 999
}
```

Run that once, then each time afterwards reduce the version by one. That will trigger the downgrade.

Screenshot_1556113273

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/WPWellSqlConfig.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package org.wordpress.android
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/WPWellSqlConfig.kt]: Can we add this whole class into folder `/src/debug/java/org/wordpress/android`? This way we'd be making use of the `productFlavors` feature and would make 100% sure the code is not even bundled within a product version.

Then, you can add a stub class in `/src/main/java/org/wordpress/android/WPWellSqlConfig.kt` that just calls the superclass `WellSqlConfig` implementation.

**Comment by maxme** [comment]
> Until the sites are loaded you'll see a message about how you don't have any sites, but once sites are loaded the My Site page populates correctly. I won't claim it's a great experience, but since it's just for us I figured it was okay :)

Got it, I'm double-checking we want to reset FluxC DB only and not the main DB.

Note: for extra precautions, we could move this to [WordPressDebug.java](https://github.com/wordpress-mobile/WordPress-Android/blob/76146508f43a2d15ffb360622e0ed5d77e175464/WordPress/src/debug/java/org/wordpress/android/WordPressDebug.java)

File: WordPress/src/main/java/org/wordpress/android/WPWellSqlConfig.kt (hunk 1 above)
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/WPWellSqlConfig.kt]: > Note: for extra precautions, we could move this to WordPressDebug.java

Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/WPWellSqlConfig.kt]: > Can we add this whole class into folder `/src/debug/java/org/wordpress/android`

Done in f73652e

File: gradle.properties-example [hunk 2]
```diff
@@ -21,6 +21,7 @@ wp.zendesk.app_id=wordpress
 wp.zendesk.domain=https://www.google.com/
 wp.zendesk.oauth_client_id=wordpress
 wp.giphy.api_key=wordpress
+wp.reset_db_on_downgrade = false
```
Comment (maxme) [inline gradle.properties-example]: +1, reminder to update private config.

**Pull request merged**